src/import/load.js

```diff
@@ -95,13 +95,22 @@ function xhrHasResponse(request) {
     request.responseText; // '' on error
 }
 
-function xhr(url, callback) {
+function xhrSetHeader(request, headers) {
+  if (headers && request.setRequestHeader) {
+    for (var name in headers) {
+      request.setRequestHeader(name, headers[name]);
+    }
+  }
+}
+
+function xhr(opts, callback) {
+  opts = typeof opts === 'string' ? { url: opts } : opts;
```

Comment:
Correct - passing an object would allow us to pass the state of the connection (e.g. cookies, headers) with each request.  sanitizeUrl() is where we can inject custom external data processing logic.